### PR TITLE
[BEAM-9170] Unify Jenkins job names related to Java 11 testing

### PR DESCRIPTION
**Pull request by mwalenia**
Previously the files with job definitions had a confusing naming scheme, as did the Jenkins jobs.
This change makes them uniform with others, thus easier to find

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by mwalenia** [comment]
R: @iemejia

**Comment by iemejia** [comment]
retest this please

**Comment by iemejia** [comment]
Run Seed Job

File: .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Direct_Java11.groovy [hunk 1]
```diff
@@ -22,7 +22,7 @@ import PostcommitJobBuilder
 
 // This job runs the suite of ValidatesRunner tests using Java 11 against the Direct
 // runner compiled with Java 8.
-PostcommitJobBuilder.postCommitJob('beam_PostCommit_Java11_ValidatesRunner_Direct',
+PostcommitJobBuilder.postCommitJob('beam_PostCommit_Java_ValidatesRunner_Direct_Java11',
         'Run Direct ValidatesRunner in Java 11', 'Direct Runner ValidatesRunner Tests for Java 11', this) {
```
Comment (iemejia) [inline .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_Direct_Java11.groovy]: Can we remove the `in` to make all names consistent in all invocations (same in the README) so it becomes `Run Direct ValidatesRunner Java 11`

File: .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_PortabilityApi_Dataflow_Java11.groovy [hunk 2]
```diff
@@ -20,7 +20,7 @@ import CommonJobProperties as commonJobProperties
 import PostcommitJobBuilder
 
 
-PostcommitJobBuilder.postCommitJob('beam_PostCommit_Java11_ValidatesRunner_PortabilityApi_Dataflow',
+PostcommitJobBuilder.postCommitJob('beam_PostCommit_Java_ValidatesRunner_PortabilityApi_Dataflow_Java11',
   'Run Dataflow PortabilityApi ValidatesRunner with Java 11', 'Google Cloud Dataflow Runner PortabilityApi ValidatesRunner Java 11 Tests', this) {
```
Comment (iemejia) [inline .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_PortabilityApi_Dataflow_Java11.groovy]: Can we remove the `with` to make all names consistent in all invocations (same in the README): `Run Dataflow PortabilityApi ValidatesRunner Java 11`

Comment (mwalenia) [reply .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_PortabilityApi_Dataflow_Java11.groovy]: Sure, I changed the phrases as well. Thanks for the suggestion. If everything else looks good, I'll squash the commits

Comment (iemejia) [reply .test-infra/jenkins/job_PostCommit_Java_ValidatesRunner_PortabilityApi_Dataflow_Java11.groovy]: Don't worry about squashing I will do on merge, seems you do not have the committer permissions set yet :S

**Comment by iemejia** [comment]
retest this please

**Comment by iemejia** [comment]
retest this please

**Comment by iemejia** [comment]
Run Seed Job

**Comment by iemejia** [comment]
Run Seed Job

**Comment by iemejia** [comment]
Run Java examples on Dataflow Java 11

**Comment by iemejia** [comment]
Run Java Portability examples on Dataflow Java 11

**Comment by iemejia** [comment]
Run Dataflow ValidatesRunner Java 11

**Comment by iemejia** [comment]
Run Dataflow PortabilityApi ValidatesRunner Java 11

**Comment by iemejia** [comment]
Run Direct ValidatesRunner Java 11

**Comment by iemejia** [comment]
Merged manually to squash the commits.

**Pull request closed without merging**